plugins/woocommerce-blocks/assets/js/base/components/combobox/index.tsx

```diff
@@ -88,72 +140,129 @@ const Combobox = ( {
-						const activeElement = isObject( controlRef.current )
-							? controlRef.current.ownerDocument.activeElement
-							: undefined;
-
-						if (
-							activeElement &&
-							isObject( controlRef.current ) &&
-							controlRef.current.contains( activeElement )
-						) {
-							return;
-						}
+			<div className="components-combobox-control components-base-control wc-block-components-combobox-control">
+				<ComboboxProvider
+					store={ store }
+					value={ searchTerm }
+					selectedValue={ selectedOption?.value || '' }
+					setValue={ ( val: string ) => {
+						startTransition( () => {
+							setSearchTerm( val );
 
-						// Try to match.
-						const normalizedFilterValue =
-							filterValue.toLocaleUpperCase();
+							if ( val?.length ) {
+								const exactLabelMatch = findExactMatchBy(
+									'label',
+									val,
+									options
+								);
 
-						// Try to find an exact match first using values.
-						const foundValue = options.find(
-							( option ) =>
-								option.value.toLocaleUpperCase() ===
-								normalizedFilterValue
+								if (
+									exactLabelMatch &&
+									exactLabelMatch.value !==
+										selectedOption?.value
+								) {
+									setSelectedOption( exactLabelMatch );
+									onChange( exactLabelMatch.value );
+								}
+							}
+						} );
+					} }
+					setSelectedValue={ ( val ) => {
+						const option = options.find(
+							( opt ) => opt.value === val
 						);
 
-						if ( foundValue ) {
-							onChange( foundValue.value );
-							return;
+						if ( option ) {
+							setSearchTerm( option.label );
+							setSelectedOption( option );
+							onChange( option.value );
 						}
+					} }
+				>
+					<div className="components-base-control__field">
+						<ComboboxLabel className="components-base-control__label">
+							{ label }
+						</ComboboxLabel>
 
-						// Fallback to a label match.
-						const foundOption = options.find( ( option ) =>
-							option.label
-								.toLocaleUpperCase()
-								.startsWith( normalizedFilterValue )
-						);
-
-						if ( foundOption ) {
-							onChange( foundOption.value );
-						}
-					}
-				} }
-				options={ options }
-				value={ value || '' }
-				allowReset={ false }
-				autoComplete={ autoComplete }
-				// Note these aria properties are ignored by ComboboxControl. When we replace ComboboxControl we should support them.
-				aria-invalid={ error?.message && ! error?.hidden }
-				aria-errormessage={ validationErrorId }
-			/>
+						<div className="components-combobox-control__suggestions-container">
+							<AriakitCombobox
+								className="components-combobox-control__input components-form-token-field__input"
+								autoComplete={ autoCompleteValue }
+								aria-invalid={ ariaInvalid }
+								aria-errormessage={ validationErrorId }
+								type="text"
+								onFocus={ () => {
+									setSearchTerm( '' );
+								} }
```

Comment:
```suggestion
								onFocus={ () => {
									store.setOpen( true );
									setSearchTerm( '' );
								} }
```

This will stop the label bug I mentioned in my last review! Feel free to apply it :)

Reply:
Thanks, i slightly adjusted this to open the menu after clearing the search term. https://github.com/woocommerce/woocommerce/pull/46943/commits/5c8105384cdbf3298d9936fba6ea805d11e852a2